### PR TITLE
Updates for travis, ruby gems

### PR DESCRIPTION
**Pull request by drenalin23**
## What
Updated gems listed in Gemfile to latest versions.
Update ruby and chefdk versions in .travis.yml
Updated rubocop and foodcritic and made some associated fixes.
Added some ubuntu 12.04 and CentOS 6.4 tests to the travis matrix so we catch failures on those platforms too.

Travis builds are now completing successfully (centos-64 builds may fail until PR 71 is merged).

### CODE REVIEW

File: recipes/apache.rb [hunk 1]
```diff
@@ -15,7 +15,7 @@
             password: node['kibana']['apache']['basic_auth_password'])
   owner node['apache']['user']
   group node['apache']['group']
-  mode 00644
+  mode 0o0644
```
Comment (mburns) [inline recipes/apache.rb]: small typo here and below

Comment (drenalin23) [inline recipes/apache.rb]: new rubocop-ness.

Comment (drenalin23) [inline recipes/apache.rb]: Offenses:

recipes/apache.rb:18:8: C: Use 0o for octal literals.
  mode 00644
       ^^^^^

Comment (mburns) [inline recipes/apache.rb]: huh. TIL.

Comment (drenalin23) [inline recipes/apache.rb]: My other thought was to switch it back to 0644 which seems fine and more succinct...

Comment (cyberflow) [inline recipes/apache.rb]: I don't know why, but tests not pass on ubuntu 12.04. look like something wrong with docker.

**Comment by cyberflow** [comment]
Please, add 'netstat' cookbook to run list for ubuntu 12.04 for test passing. You can check my try to fix it https://github.com/cyberflow/chef-kibana/commit/d0b6e94421a288569557760aa60e94b295e12ca4

**Comment by drenalin23** [comment]
@cyberflow That appears to work!  I tested locally with kitchen test on those builds and then those have run successfully on a new push of those commits to my branch.  When they all finish I'll merge this in.

**Pull request merged**